en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-300-to-310.md

```diff
@@ -0,0 +1,982 @@
+# Upgrading API Manager from 3.1.0 to 3.1.0
+
+The following information describes how to upgrade your API Manager server **from APIM 3.1.0 to 3.1.0**.
+
+!!! note
+    Before you follow this section, see [Upgrading Process](../upgrading-wso2-api-manager/upgrading-process.md) for more information.
+
+!!! attention "Before you Begin"
+    This release is a WUM-only release. This means that there are no manual patches. Any further fixes or latest updates for this release can be updated through the WSO2 Update Manager (WUM).
+
+    -   **If you are upgrading to this version, in order to use this version in your production environment** , use the WSO2 Update Manager and get the latest available updates for WSO2 API Manager 3.1.0. For more information on how to do this, see [Updating WSO2 Products](https://docs.wso2.com/display/updates/Using+WSO2+Update+Manager).
+
+Follow the instructions below to upgrade your WSO2 API Manager server **from WSO2 API-M 3.0.0 to 3.1.0**.
+
+### Preparing for Migration
+#### Disabling versioning in the registry configuration
+
+If there are frequently updating registry properties, having the versioning enabled for registry resources in the registry can lead to unnecessary growth in the registry related tables in the database. To avoid this, versioning has been disabled by default in API Manager 3.1.0.
+
+Therefore, when migrating to API Manager 3.1.0, it is **required** to turn off the registry versioning in your
+current API Manager 3.0.0 version and run the below scripts against **the database that is used by the registry**.
+
+!!! note "NOTE"
+    Alternatively, it is possible to turn on registry versioning in API Manager 3.1.0 and continue. But this is
+    highly **NOT RECOMMENDED** and these configurations should only be changed once.
+
+!!! info "Turning off registry versioning in your current API-M and running the scripts"
```

Comment:
Shall we add the deployment.toml configuration to enable registry versioning here. I think that would be more appropriate.

Reply:
Why would we need a registry versioning enabling config as we are not recommending it? Better to remove this part rather than adding the enabling config WDYT?

Reply:
Fixed in 38ea33a145c878e71ad0aa9069d0b137f4212c09

Reply:
What I mentioned in the earlier comment is, we have added `registry.xml `configuration to turn off registry versioning and we need to change it with `toml` configuration.
Anyway, since we disabled registry versioning by default in 3.0.0, +1 to change it